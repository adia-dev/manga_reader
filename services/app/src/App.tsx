import { Outlet } from 'react-router-dom'
import './App.css'
import Header from './components/Header'
import HelpBox from './components/HelpBox'
import Searchbar from './components/Searchbar'


function App() {

  return (

    <div className="bg-dark-primary w-screen relative overflow-hidden" id="app" data-testid="app">
      <Searchbar />
      <Header />
      <Outlet />
      <HelpBox />
      {/* <Debug /> */}
    </div>

  )
}

export default App
