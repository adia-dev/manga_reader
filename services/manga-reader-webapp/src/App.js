import "./App.css";
import Home from "./pages/Home";

function App() {
  return (
    <div className="">
      <Home />
    </div>
  );
}

export default App;
