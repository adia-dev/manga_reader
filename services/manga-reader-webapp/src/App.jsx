import { useEffect, useState } from "react";
import "./App.css";
import Searchbar from "./components/Searchbar";
import Home from "./pages/Home";
import Manga from "./pages/Manga";
import mangas from "./data/mangas.json";
import Header from "./components/Header";

function App() {

  const [searchBarOpened, setSearchBarOpened] = useState(false);

  useEffect(() => {
    // when I press cmd + k, the search bar should open
    const handleKeyDown = (e) => {
      if ((e.metaKey || e.ctrlKey) && e.key === "k") {
        setSearchBarOpened(true);
      }

      if (e.key === "Escape") {
        setSearchBarOpened(false);
      }
    }

    window.addEventListener("keydown", handleKeyDown);
  }, [searchBarOpened]);


  return (
    <div className="h-screen" data-testid="app">
      <Home setSearchBarOpened={setSearchBarOpened} />
      <Header setSearchBarOpened={setSearchBarOpened} />
      <Manga manga={mangas[0]} />
      <Searchbar searchBarOpened={searchBarOpened} setSearchBarOpened={setSearchBarOpened} />

    </div>
  );
}

export default App;
