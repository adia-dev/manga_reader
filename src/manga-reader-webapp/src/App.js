import "./App.css";

function App() {
  return (
    <div className="">
      <h1 className="text-3xl font-bold underline">Hello world!</h1>
    </div>
  );
}

export default App;
