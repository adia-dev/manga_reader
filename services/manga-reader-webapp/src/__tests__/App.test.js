import { render, screen } from "@testing-library/react";
import App from "../App";

it("Checks if the app renders", () => {
  render(<App />);
  const app = screen.getByTestId("app");
  expect(app).toBeInTheDocument();
});
