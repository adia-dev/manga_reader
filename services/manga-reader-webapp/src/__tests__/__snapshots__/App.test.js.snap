// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Hello ! on the App 1`] = `
<div
  className=""
>
  <div>
    Home
  </div>
</div>
`;

exports[`Renders Hello World ! on the App 1`] = `
<div
  className=""
>
  <div>
    Home
  </div>
</div>
`;

exports[`Renders Hello World! on the App 1`] = `
<div
  className=""
>
  <h1
    className="text-3xl font-bold underline"
  >
    Hello world!
  </h1>
</div>
`;

exports[`Renders Home on the App 1`] = `
<div
  className=""
>
  <div>
    Home
  </div>
</div>
`;

exports[`Renders Homs on the App 1`] = `
<div
  className=""
>
  <div>
    Home
  </div>
</div>
`;
